clang/test/Frontend/sycl-propagate-aspect-warning.cpp

```diff
@@ -0,0 +1,97 @@
+// RUN: %clang_cc1 -internal-isystem %S/Inputs -fsycl-is-device -mconstructor-aliases -no-opaque-pointers -emit-codegen-only -verify %s
+
+// Tests for warnings when propagated aspects do not match the aspects available
+// in a function, as specified through the 'sycl::device_has' attribute.
+
+#include "sycl.hpp"
+
+// expected-note@+1 4 {{propagated from call to function 'Struct1::Struct1()'}}
+struct [[__sycl_detail__::__uses_aspects__(sycl::aspect::fp16)]] Struct1 {
+  int a = 0;
+};
+
+// expected-note@+1 4 {{propagated from call to function 'func3(int, int, int)'}}
+int func3(int a, int b, int c) {
+  Struct1 s;
+  s.a = 1;
+  return s.a;
+}
+
+// expected-note@+1 4 {{propagated from call to function 'func4(int, int)'}}
```

Comment:
"propagated to" just sounds weird IMO. Can't we just say "in func4(int, int)" I think that's more in line with how diagnostic trace is handled now.

 @Fznamznon thoughts?

Reply:
I'm fine with both variants.

Reply:
@steffenlarsen I don't fully understand the intention of this note diagnostic. What do you mean by 'propagated to' ? Are you just trying to point to the function declarations after emitting the warning on the kernel? If so, can you use this note instead - `note_entity_declared_at`.